### DISCUSSION TITLE
deepin-terminal增加加粗下划线光标选项

### DISCUSSION THREAD

**Discussion by liwl1991** [linuxdeepin/developer-center#4030] [Features Request & Ideas | 特性请求 & 头脑风暴]
目前的终端选项中，只有粗竖线，细下划线，细竖线，3个选项。
没有粗下划线的选项。
在使用vim编辑文本的时候，选择光标为细下划线时，视觉上会无法快速定位光标，很不方便。
我已经实现了加粗下划线的编码，在提交pr之前，想确认下这样的编码是否符合产品需求？

**Comment by liwl1991** [comment]
@ArchieMeng @meiyixiang

**Reply by ArchieMeng** [reply]
我看可以，看看产品怎么说。

**Comment by BLumia** [comment]
是通过修改 vendor 的 qtermwidget 实现的？如果是的话，要不要考虑一并把修改 upstream 到 qtermwidget 上游？

**Reply by liwl1991** [reply]
> 是通过修改 vendor 的 qtermwidget 实现的？如果是的话，要不要考虑一并把修改 upstream 到 qtermwidget 上游？

对的，是修改qtermwidget的代码实现的，我增加了一个选项BoldUnderlineCursor，像素增加到4
如果往上游提交？没操作过。是直接fork那个库？

**Reply by ArchieMeng** [reply]
> 对的，是修改qtermwidget的代码实现的，我增加了一个选项BoldUnderlineCursor，像素增加到4
> 如果往上游提交？没操作过。是直接fork那个库？

是的，fork qtermwidget下来，然后加上你做的修改，可以的话和qterminal 一起编验证下修改效果。

**Reply by BLumia** [reply]
> 如果往上游提交？没操作过。是直接fork那个库？

上游是 https://github.com/lxqt/qtermwidget 。目前 deepin-terminal 自带的其实是魔改的版本。个人建议是如果这个功能确实比较合适的话，可以向上游推进，这个推进的过程中也可以以库的维护的角度来看这种接口是否适合添加/如何添加，推进成功的话，基于它的软件（deepin-terminal，qterminal）可以直接受益。

**Reply by ArchieMeng** [reply]
> 
> 对的，是修改qtermwidget的代码实现的，我增加了一个选项BoldUnderlineCursor，像素增加到4 如果往上游提交？没操作过。是直接fork那个库？

我发现如果要往上游推的话，这个commit应该要做点修改(updateCursor方法需要改)。可以参考下这个commit. https://github.com/ArchieMeng/dtermwidget/commit/462189bd3282bfac5ada42af88e81a5eb8a861ce

**Reply by liwl1991** [reply]
好的，我可能这周六（2023.04.22）向上提交了

**Reply by ArchieMeng** [reply]
> 好的，我可能这周六（2023.04.22）向上提交了

好像还有另一个问题。输入文本的时候，前一个光标会残留，然后新光标会变成原来的细的那个。闪一下才恢复正常。

**Reply by liwl1991** [reply]
我没有发现类似的问题呢？你是在虚拟机里面的吗？会是虚拟机反应迟钝导致？

**Reply by ArchieMeng** [reply]
不是。你可以试试用ArchieMeng/deepin-terminal的update-terminalwidget分支代码看看。这个是用较新的上游qtermwidget魔改的。

**Reply by liwl1991** [reply]
啊这，但是我在master上提交的那个pr，其目录下的3rdparty是老版本的魔改是吗？你准备切换到新的分支，那我需要修复这样的问题，是不是？

**Reply by ArchieMeng** [reply]
是的．虽然不影响deepin-terminal的主线，但这会是你推上游要遇到的问题，你也可以先拿qterminal试试看看，直接这样用有没有问题．当然，这也是现在更新qtermwidget遇到了的问题，我也在看着．

**Reply by liwl1991** [reply]
行的，那我有时间先去试试看看qterminal吧，如果在qterminal上没有问题，那才能推的，不然推上去也是bug

**Comment by meiyixiang** [comment]
没有问题，提升了使用过程中的用户体验。

**Comment by Aninana** [comment]
我也来了！试试？
